src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/ProtobufOtlpGrpcExportClient.cs

```diff
@@ -27,49 +25,34 @@ private static readonly ExportClientGrpcResponse DefaultExceptionExportClientGrp
             grpcStatusDetailsHeader: null);
 
     public ProtobufOtlpGrpcExportClient(OtlpExporterOptions options, HttpClient httpClient, string signalPath)
+        : base(options, httpClient, signalPath)
     {
-        Guard.ThrowIfNull(options);
-        Guard.ThrowIfNull(httpClient);
-        Guard.ThrowIfNull(signalPath);
-        Guard.ThrowIfInvalidTimeout(options.TimeoutMilliseconds);
-
-        Uri exporterEndpoint = options.Endpoint.AppendPathIfNotPresent(signalPath);
-        this.Endpoint = new UriBuilder(exporterEndpoint).Uri;
-        this.Headers = options.GetHeaders<Dictionary<string, string>>((d, k, v) => d.Add(k, v));
-        this.HttpClient = httpClient;
     }
 
-    internal HttpClient HttpClient { get; }
+    internal override MediaTypeHeaderValue MediaTypeHeader => MediaHeaderValue;
 
-    internal Uri Endpoint { get; set; }
-
-    internal IReadOnlyDictionary<string, string> Headers { get; }
-
-    internal int TimeoutMilliseconds { get; }
+    internal override bool RequireHttp2 => true;
 
     /// <inheritdoc/>
-    public ExportClientResponse SendExportRequest(byte[] buffer, int contentLength, DateTime deadlineUtc, CancellationToken cancellationToken = default)
+    public override ExportClientResponse SendExportRequest(byte[] buffer, int contentLength, DateTime deadlineUtc, CancellationToken cancellationToken = default)
     {
         try
         {
             using var httpRequest = this.CreateHttpRequest(buffer, contentLength);
             using var httpResponse = this.SendHttpRequest(httpRequest, cancellationToken);
 
-            try
-            {
-                httpResponse.EnsureSuccessStatusCode();
-            }
-            catch (HttpRequestException)
-            {
-                throw;
```

Comment:
@rajkumar-rangaraj I removed this try/catch because it just seems to throw always. Can you think of a reason for this, did I miss something?

Reply:
This is similar to our existing implementation - https://github.com/open-telemetry/opentelemetry-dotnet/blob/88d2ad60bb1b6ea0162eb8fa3cd2d006b17c4c55/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/BaseOtlpHttpExportClient.cs#L60